src/server.lisp

```diff
@@ -0,0 +1,184 @@
+;;;; server.lisp
+;;;;
+;;;; Author: Eric Peterson
+;;;;
+;;;; Lisp mimic of the python JSON RPC Server.
+
+(in-package #:rpcq)
+
+
+(defun make-dispatch-table ()
+  (make-hash-table :test #'equal))
+
+(defun dispatch-table-add-handler (dispatch-table f
+                                   &key
+                                     (name (string f)))
+  "Add a function to a dispatch table, for use by an RPCQ server.  By default, a symbol passed in for F will be automatically converted into the name of F used in the dispatch table.  To manually specify a name (or to provide a name for a non-symbol value of F), use the keyword argument :NAME."
+  (setf (gethash (sanitize-name name) dispatch-table) f)
+  nil)
+
+
+(defun %pull-raw-request (receiver)
+  "Pulls a ZMQ request over the RECEIVER socket.  Returns a VALUES triple:
+
+* IDENTITY-FRAME: array of (INTEGER 256) that describes the intended recipient of a reply to this request.
+* NULL-FRAME?:    boolean indicating whether the recipient expects an additional null frame after the identity frame. (This is the case for REQ-type clients.)
+* PAYLOAD:        array of (INTEGER 256) that houses the raw request."
+  ;; ZeroMQ requests come in one of two flavors:
+  ;;
+  ;;   (1) identity frame, null frame, data frame
+  ;;   (2) identity frame, data frame
+  ;;
+  ;; whichever we get, we also have to reply in the same way, so we track the
+  ;; format in addition to the data.
+  (flet ((unpack-foreign-msg-to-bytes (msg)
+           (cffi:foreign-array-to-lisp (pzmq:msg-data msg)
+                                       `(:array :unsigned-char ,(pzmq:msg-size msg)))))
```

Comment:
Perhaps we should consider `:uint8` instead of `:unsigned-char`. See my previous question about `:unsigned-char`.

Reply:
Will this cause weird type errors for `string` data from Lisp to C?